test/test_database.py

```diff
@@ -132,7 +131,8 @@ def test_create_collection(self):
         db = Database(self.client, "pymongo_test")
 
         db.test.insert_one({"hello": "world"})
-        self.assertRaises(CollectionInvalid, db.create_collection, "test")
+        if not client_context.is_topology_type(["sharded"]):
+            self.assertRaises(OperationFailure, db.create_collection, "test")
```

Comment:
We should also test the behavior of `db.create_collection` on an existing collection on a sharded cluster. For example, there should be an `else:` here or we should have a test called `test_create_collection_existing`